src/Faker/ValidGenerator.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+namespace Faker;
+
+/**
+ * Proxy for other generators, to return only valid values. Works with
+ * Faker\Generator\Base->valid()
+ */
+class ValidGenerator
+{
+    protected $generator;
+    protected $validator;
+    protected $maxRetries;
+
+    /**
+     * @param Generator $generator
+     */
+    public function __construct(Generator $generator, $validator = null, $maxRetries = 10000)
```

Comment:
What about forcing `$validator` to be a `\Closure`?
Because without forcing it, functions not returning a boolean could be used.